syntax = "proto3";

package audio_service;

// Mensajes
message Metadata {
    string file_name = 1;
    int32 duration_seconds = 2;
    int32 sample_rate = 3;
    int32 channels = 4;
    string codec = 5;
}

message AudioChunk {
    bytes chunk_data = 1;
    int32 sequence_number = 2;
}

message MetadataRequest {
    string file_name = 1;
}

message MP3List {
    repeated string file_names = 1;
}

message StreamRequest {
    string file_name = 1;
}

message Empty {}

// Servicios
service MP3Service {
    rpc ListAvailableMP3 (Empty) returns (MP3List);
}

service AudioStreamService {
    rpc StreamAudio (StreamRequest) returns (stream AudioChunk); // <-- NOMBRE CAMBIADO
    rpc GetMetadata (MetadataRequest) returns (Metadata);        // <-- NOMBRE CAMBIADO
}

service MetadataService {
    rpc GetMetadata (MetadataRequest) returns (Metadata);
}
